src/main/java/com/iexec/worker/sms/SmsService.java

```diff
@@ -47,81 +49,68 @@ public SmsService(SmsClient smsClient,
         this.workerConfigurationService = workerConfigurationService;
     }
 
+    @Retryable(value = FeignException.class)
     public boolean fetchTaskSecrets(ContributionAuthorization contributionAuth) {
-        String hash = HashUtils.concatenateAndHash(contributionAuth.getWorkerWallet(),
-                contributionAuth.getChainTaskId(), contributionAuth.getEnclaveChallenge());
+        String chainTaskId = contributionAuth.getChainTaskId();
+        SmsRequest smsRequest = buildSmsRequest(contributionAuth);
 
-        Sign.SignatureData workerSignature = Sign.signPrefixedMessage(
-                BytesUtils.stringToBytes(hash), credentialsService.getCredentials().getEcKeyPair());
+        SmsSecretResponse smsResponse = smsClient.getTaskSecrets(smsRequest);
 
-        SmsSecretRequestBody smsSecretRequestBody = SmsSecretRequestBody.builder()
-                .chainTaskId(contributionAuth.getChainTaskId())
-                .workerAddress(contributionAuth.getWorkerWallet())
-                .enclaveChallenge(contributionAuth.getEnclaveChallenge())
-                .coreSignature(contributionAuth.getSignature().getValue())
-                .workerSignature(new Signature(workerSignature).getValue())
-                .build();
+        if (smsResponse == null) {
+            log.error("Received null response from SMS [chainTaskId:{}]", chainTaskId);
+            return false;
+        }
 
-        Optional<TaskSecrets> oTaskSecrets = getTaskSecrets(smsSecretRequestBody);
+        if (!smsResponse.isOk()) {
+            log.error("An error occurred while getting task secrets [chainTaskId:{}, errorMessage:{}]",
+                    chainTaskId, smsResponse.getErrorMessage());
+            return false;
+        }
+
+        TaskSecrets taskSecrets = smsResponse.getData().getSecrets();
 
-        if (!oTaskSecrets.isPresent()) {
-            log.error("Could not call SMS to get secrets, aborting [chainTaskId:{}]",
-                    contributionAuth.getChainTaskId());
+        if (taskSecrets == null) {
+            log.error("Received null secrets object from SMS [chainTaskId:{}]", chainTaskId);
             return false;
         }
 
-        saveSecrets(contributionAuth.getChainTaskId(), oTaskSecrets.get());
+        saveSecrets(chainTaskId, taskSecrets);
         return true;
     }
 
-    public void saveSecrets(String chainTaskId, TaskSecrets taskSecrets) {
+    @Recover
+    private boolean fetchTaskSecrets(FeignException e, ContributionAuthorization contributionAuth) {
+        log.error("Failed to get task secrets from SMS [chainTaskId:{}, attempts:3]",
+                contributionAuth.getChainTaskId());
+        e.printStackTrace();
+        return false;
+    }
 
+    public void saveSecrets(String chainTaskId, TaskSecrets taskSecrets) {
         SmsSecret datasetSecret = taskSecrets.getDatasetSecret();
```

Comment:
So here they're all encryptedSecrets?

Reply:
No, they aren't. We just save what we received.

Reply:
The secret is readable to anyone ?

Reply:
Yes, they are, otherwise, it's TEE workflow.